### ISSUE TITLE
Processing legacy "View Engine" libraries:

### ISSUE THREAD

**Issue by wilson3q** [stephanrauh/ngx-extended-pdf-viewer#1033]
Not a bug. Just a little enhancement request.

While building with angular 13, the following messages are printed:

```
Processing legacy "View Engine" libraries:

- ngx-extended-pdf-viewer [es2015/esm2015] (https://github.com/stephanrauh/ngx-extended-pdf-viewer)

Encourage the library authors to publish an Ivy distribution.
```

Please help to migrate to ivy if you have free time.
The best PDF tools for angular!

Thanks.

**Comment by stephanrauh** [comment]
I'd like to do that. I just don't have a clue what I need to do. Maybe you can help me.

As far as I understand, the warning says I have to drop the pre-ivy support. But that means I have to drop support for Angular 11 and below. At the moment, I don't want to do that.

When Angular 12 was published, I updated the library so it supports both Ivy and pre-Ivy Angular versions. I'm not sure if I did it wrong, or if Angular 14 is going to stop supporting libraries that are compatible with Angular 10 or 11.

Do you happen to know something about the topic?

**Comment by wilson3q** [comment]
Very simple. According to [https://blog.angular.io/upcoming-improvements-to-angular-library-distribution-76c02f782aa4](url)

```
If you’re a library developer and you’re not ready to move to Ivy just yet, in v12 update your tsconfig.lib.prod.jsonto:
/* To learn more about this file see: https://angular.io/config/tsconfig. */
{
  "extends": "./tsconfig.lib.json",
  "compilerOptions": {
    "declarationMap": false
  },
  "angularCompilerOptions": {
    "enableIvy": false
  }
}
```

**Comment by stephanrauh** [comment]
I already configured the project like this. I didn't disable the declaration map, but I'm not sure it's got anything to do with Ivy. I guess I'll test it tomorrow.

**Comment by LAlves91** [comment]
Hi @stephanrauh !

I think in order to make ngx-extended-pdf-viewer Ivy optimized (with 'partial' as the compilation mode), View Engine support must be dropped.

The only way I see to support both is using a multi-release pattern: launching a v10.x with PDF.js related enhancements/bugfixes but keeping 'enableIvy' false, and making v11 full Ivy ('enableIvy' true and 'compilationMode' partial). That would complicate the development process a little, though.

**Comment by stephanrauh** [comment]
Hm, Ivy has become the default in Angular 9. So everybody should have migrated in the meantime. So dropping Ivy support might be a good idea. I'll check tomorrow if my company project uses Ivy or not. The project a bit conservative, so it might be a good project to verify my theory.

**Comment by stephanrauh** [comment]
I've decided to drop pre-Ivy support with version 11. Any objections?

**Comment by stephanrauh** [comment]
I've just published version 11.0.0-alpha.8. It drops support for the old view engine.

**Comment by wilson3q** [comment]
Thanks a lot! Great support!
